lib/cocoapods-core/specification/json.rb

```diff
@@ -29,6 +29,7 @@ def to_hash
         unless subspecs.empty?
           hash['subspecs'] = subspecs.map(&:to_hash)
         end
+        hash['test_specification'] = test_specification?
```

Comment:
The json store should be test type, otherwise we lose data in the transfer I think?